macros/src/main/scala/MacroCompiler.scala

```diff
@@ -561,9 +564,32 @@ class MacroCompilerPass(mems: Option[Seq[Macro]],
         // in the 'circuit'.
         (mems foldLeft c.modules){ (modules, mem) =>
 
+        val sram = mem.src
+        def groupMatchesMask(group: SRAMGroup, mem:SRAMMacro): Boolean = {
+          val memMask = mem.ports map (_.maskGran) find (_.isDefined) map (_.get)
+          val libMask = group.ports map (_.maskGran) find (_.isDefined) map (_.get)
+          (memMask, libMask) match {
+            case (_, Some(1)) => true
```

Comment:
Isn't this a subset of cases 2 and 4 below?

Reply:
Good catch thanks!